### PR TITLE
feat: start thread page

### LINKED ISSUE #18: Thread page
Thread page with message list

### PR DESCRIPTION
**Pull request by fknop**
Current progress:

Closes #18

### CODE REVIEW

File: forum/models.py [hunk 1]
```diff
@@ -28,7 +28,17 @@ def is_public_professor(self):
         return self.professor is not None
 
     def messages(self):
-        return Message.objects.filter(thread=self).order_by("created_date")
+        return Message.objects.select_related('author').filter(thread=self, parent_message=None).order_by("created_date")
+
+    def messages_with_replies(self):
```
Comment (fknop) [inline forum/models.py]: Will probably be deleted

File: forum/models.py [hunk 2]
```diff
@@ -55,24 +65,35 @@ class Message(models.Model):
...
+        for message in Message.objects.filter(thread=self.thread, parent_message=self).order_by('created_date'):
             replies.append(message)
 
         return replies
 
     def all_replies(self):
-        replies = {self: []}
-        for message in Message.objects.filter(parent_message=self).order_by("created_date"):
-            replies[self].append(message.all_replies())
+        replies = []
```
Comment (fknop) [inline forum/models.py]: Will probably be deleted

File: forum/tests.py [hunk 3]
```diff
@@ -10,6 +11,24 @@
 
 
 class ThreadModelTest(TestCase):
+
+    def testInvalidThread(self):
```
Comment (mlosicki) [inline forum/tests.py]: Change method name to something clearer

File: forum/tests.py [hunk 4]
```diff
@@ -98,62 +118,22 @@ def testReplies(self):
...
-        self.assertEquals(messages[1].id, second_message.id)
+        self.assertEquals(messages[0], first_message)
 
         replies = first_message.replies()
         self.assertEquals(replies[0], second_message)
 
-        all_replies = first_message.all_replies()
-        self.assertEquals(all_replies[first_message], [{second_message: []}])
-
-    def testAllReplies(self):
```
Comment (mlosicki) [inline forum/tests.py:118]: Removed?

File: forum/templatetags/forum_tags.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+@register.inclusion_tag('forum/message.haml')
+def message_partial(message, user, level=0):
+    return {
+        "message": message,
+        "level": level + 1,
+        "user": user
+    }
+
+
+@register.filter
```
Comment (mlosicki) [inline forum/templatetags/forum_tags.py:13]: Is this the right way to do it? Seems doable in CSS only

File: forum/tests.py (hunk 4 above)
Comment (fknop) [reply forum/tests.py:118]: Yes. Method was removed.

**Review by mlosicki — APPROVED** [review]

**Pull request merged**